### PR TITLE
Automated executable build and artifact archiving

### PR DESCRIPTION
**Pull request by cha2251**
- On pushes to main, a build will be triggered and the artifact saved off
- No CD yet, git has a release function, but for now keep manual
- Created main class to test that build system could access all subsystems

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,9 +3,6 @@ __pycache__/
 *.py[cod]
 *$py.class
 
-# C extensions
-*.so
-
```
Comment (dsp9049) [inline .gitignore:8]: This might make a merge conflict with your other pull request:
https://github.com/cha2251/rit-player-piano/pull/11/files

Comment (cha2251) [reply .gitignore:8]: I branched that one off of this, so it should be good. But if not I'll fix the conflict in that one

File: README.md [hunk 2]
```diff
@@ -20,4 +20,12 @@ https://cha2251.github.io/rit-player-piano/
...
 - Follow test discovery syntax when creating new tests 
- - https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#test-discovery
\ No newline at end of file
+ - https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#test-discovery
+
+# Building Executeable
+- Start virtual environment
+- Windows:
+    - In top level directory run: `pyinstaller.exe src\common\main.py --clean --onefile`
+        - Fix permission erros with: `chmod -R -c u+rwx .`
```
Comment (DavidTheFighter) [inline README.md:29]: Why is this in the Windows section? Windows is inferior and doesn't have `chmod`. Also typo in 'erros'.

**Review by DavidTheFighter — APPROVED** [review]
Besides the one line in the readme it looks good! Tested on both my Linux and Windows machines and both build and run just fine.

File: README.md (hunk 2 above)
Comment (cha2251) [reply README.md:29]: I.. uh.. it works on my machine?? Now that I think about it I'm a little concerned it does tho

**Pull request merged**